lib/uwb/protocols/fira/UwbOobConversions.cxx

```diff
@@ -0,0 +1,126 @@
+         return config.GetKeyRotationRate();
+     } },
+    { UwbApplicationConfigurationParameterType::MacAddressMode, [](const UwbConfiguration& config, DeviceType deviceType) -> std::optional<UwbApplicationConfigurationParameterValue> {
+         return config.GetMacAddressMode();
+     } },
+    { UwbApplicationConfigurationParameterType::MaxRangingRoundRetry, [](const UwbConfiguration& config, DeviceType deviceType) -> std::optional<UwbApplicationConfigurationParameterValue> {
+         return config.GetMaxRangingRoundRetry();
+     } },
+    { UwbApplicationConfigurationParameterType::UwbInitiationTime, [](const UwbConfiguration& config, DeviceType deviceType) -> std::optional<UwbApplicationConfigurationParameterValue> {
+         return config.GetUwbInitiationTime();
+     } },
+    { UwbApplicationConfigurationParameterType::HoppingMode, [](const UwbConfiguration& config, DeviceType deviceType) -> std::optional<UwbApplicationConfigurationParameterValue> {
+         return config.GetHoppingMode();
+     } },
+    // {  UwbApplicationConfigurationParameterType::ResultReportConfig, []( const UwbConfiguration & config) -> std::optional< UwbApplicationConfigurationParameterValue> {
+    //      return config.GetResultReportConfigurations();
+    //  } },
+
+    // params with different names
+    // {  UwbApplicationConfigurationParameterType::RangingRoundUsage, []( const UwbConfiguration & config) -> std::optional< UwbApplicationConfigurationParameterValue> {
+    //      return config.GetRangingMethod();
+    //  } },
+    { UwbApplicationConfigurationParameterType::DeviceMacAddress, [](const UwbConfiguration& config, DeviceType deviceType) -> std::optional<UwbApplicationConfigurationParameterValue> {
+         auto mode = config.GetMacAddressMode();
+         if (not mode) {
+             return std::nullopt;
+         }
+         if (deviceType == DeviceType::Controller) {
+             return config.GetControllerMacAddress();
+         } else {
+             if (mode == uwb::UwbMacAddressType::Short) {
+                 return config.GetControleeShortMacAddress();
+             } else {
+                 // TODO what do we do here
+                 return std::nullopt;
+             }
+         }
+         return std::nullopt;
+     } },
+    { UwbApplicationConfigurationParameterType::DestinationMacAddresses, [](const UwbConfiguration& config, DeviceType deviceType) -> std::optional<UwbApplicationConfigurationParameterValue> {
+         auto mode = config.GetMacAddressMode();
+         if (not mode) {
+             return std::nullopt;
+         }
+         if (deviceType == DeviceType::Controller) {
+             if (mode == uwb::UwbMacAddressType::Short) {
+                 return config.GetControleeShortMacAddress(); // TODO this should reflect the possibility for multiple peers
```

Comment:
This might not be necessary. If our current understanding is accurate, then we'll expect many `UwbSessionData` (with `UwbConfiguration` in them) to be provided via OOB, so we'd call `UwbSession::Configure` for each of them after converting them with this code, and each of them should only contain a single address for the controllee. Eg.

```
DeviceType deviceType = DeviceType::Controller; // from OOB "profile"
UwbSession uwbSession = uwbDevice.CreateSession(deviceType);
foreach (UwbSessionData from OOB /* = N */) {
    auto applicationConfigurationParameters = GetUciConfigParams(UwbSessionData.uwbConfiguration, deviceType);
    uwbSession.Configure(uwbSessionData.sessionId, std::move(applicationConfigurationParameters));
}
// uwbSession is now configured for 'N' sets of parameters for 'N' controlees
```

I think that when the host is a controller, it will set a lot of the application configuration parameters itself, whereas when the host is a controlee, it gets most of the parameters from the controller, and there will likely only be one `UwbSessionData` from OOB.